api/nodes/views.py

```diff
@@ -28,7 +28,82 @@ def get_node(self):
         return obj
 
 
-class NodeList(generics.ListCreateAPIView, ODMFilterMixin):
+class NodeIncludeMixin(object):
+
+    def __init__(self, additional_query_params=None, user=None):
+        self.query_params = additional_query_params
+        self.auth = Auth(user)
+
+    def get_relationship_meta_data(self, obj, object_type):
+        included = False
+        if not self.query_params:
+            self.query_params = self.request.query_params
+            self.auth = Auth(self.request.user)
+        query_params = self.request.query_params if hasattr(self, 'request') else self.query_params
+        if 'include' in query_params:
+            for param in self.query_params['include'].split(','):
+                if param not in ['children', 'contributors', 'pointers', 'registrations']:
+                    raise ValidationError('{} is an invalid key.'.format(param))
+                if param == object_type:
+                    included = True
+        ret = None
+        if object_type == 'children':
+            ret = self.get_node_data(obj, included)
+        if object_type == 'contributors':
+            ret = self.get_contrib_data(obj, included)
+        if object_type == 'pointers':
+            ret = self.get_pointers_data(obj, included)
+        if object_type == 'registrations':
+            ret = self.get_registration_data(obj, included)
+        return ret
+
+    def get_node_data(self, obj, included):
+        ret = {}
+        nodes = [node for node in obj.nodes if node.can_view(self.auth) and node.primary]
+        ret['count'] = len(nodes)
+        if included:
+            ret['data'] = []
+            for node in nodes:
+                serialized_node = NodeSerializer(node)
+                ret['data'].append(serialized_node.data['data'])
+        return ret
+
+    def get_contrib_data(self, obj, included):
+        ret = {}
+        contributors = obj.contributors
+        ret['count'] = len(contributors)
+        if included:
+            ret['data'] = []
+            for contributor in contributors:
+                contributor.bibliographic = obj.get_visible(contributor)
+                serialized_contributor = ContributorSerializer(contributor)
+                ret['data'].append(serialized_contributor['data'])
+        return ret
+
+    def get_pointers_data(self, obj, included):
+        ret = {}
+        pointers = obj.nodes_pointer
+        ret['count'] = len(pointers)
+        if included:
+            ret['data'] = []
+            for pointer in pointers:
+                serialized_pointer = NodePointersSerializer(pointer)
+                ret['data'].append(serialized_pointer.data['data'])
+        return ret
+
+    def get_registration_data(self, obj, included):
```

Comment:
nodes:nodes-registrations already has a get_queryset() method. If it also had a get_queryset_metadata() function that did the count and anything else you might want for metainformation on a collection, then I'm pretty sure you could make one function that grabbed the queryset and the queryset_metadata for any object that it's being applied to rather than having one function for pointers, one for registrations, and so on. You could have a default get_queryset_metadata() that would just return an empty dictionary or None (None might be better so you don't bother rendering the meta section if it's None).